tests/client/test_auth.py

```diff
@@ -1,44 +1,56 @@
 import hashlib
-import json
 import os
 import typing
 
+import httpcore
 import pytest
 
 from httpx import (
     URL,
     AsyncClient,
     Auth,
     DigestAuth,
+    Headers,
     ProtocolError,
     Request,
     RequestBodyUnavailable,
     Response,
 )
-from httpx._config import CertTypes, TimeoutTypes, VerifyTypes
-from httpx._dispatch.base import AsyncDispatcher
+from httpx._content_streams import ContentStream, JSONStream
 
 
-class MockDispatch(AsyncDispatcher):
-    def __init__(self, auth_header: str = "", status_code: int = 200) -> None:
+def get_header_value(headers, key, default=None):
+    lookup = key.encode("ascii").lower()
+    for header_key, header_value in headers:
+        if header_key.lower() == lookup:
+            return header_value.decode("ascii")
+    return default
+
+
+class MockDispatch(httpcore.AsyncHTTPProxy):
+    def __init__(self, auth_header: bytes = b"", status_code: int = 200) -> None:
         self.auth_header = auth_header
         self.status_code = status_code
 
-    async def send(
+    async def request(
         self,
-        request: Request,
-        verify: VerifyTypes = None,
-        cert: CertTypes = None,
-        timeout: TimeoutTypes = None,
-    ) -> Response:
-        headers = [("www-authenticate", self.auth_header)] if self.auth_header else []
-        body = json.dumps({"auth": request.headers.get("Authorization")}).encode()
-        return Response(
-            self.status_code, headers=headers, content=body, request=request
+        method: bytes,
+        url: typing.Tuple[bytes, bytes, int, bytes],
+        headers: typing.List[typing.Tuple[bytes, bytes]],
+        stream: ContentStream,
+        timeout: typing.Dict[str, typing.Optional[float]] = None,
+    ) -> typing.Tuple[
+        bytes, int, bytes, typing.List[typing.Tuple[bytes, bytes]], ContentStream
+    ]:
+        authorization = get_header_value(headers, "Authorization")
+        response_headers = (
+            [(b"www-authenticate", self.auth_header)] if self.auth_header else []
         )
+        response_stream = JSONStream({"auth": authorization})
+        return b"HTTP/1.1", self.status_code, b"", response_headers, response_stream
 
 
-class MockDigestAuthDispatch(AsyncDispatcher):
+class MockDigestAuthDispatch(httpcore.AsyncHTTPProxy):
```

Comment:
Why does this have to be an `HTTPProxy` now? (Is it an oversight?)

Reply:
Thanks! Oughta be `httpcore.AsyncHTTPTransport` yup.